mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -236,13 +235,8 @@ private UDTValue toUDT(MessageAttachment messageAttachment) {
     }
 
     public Flux<MessageResult> retrieveMessages(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, Limit limit) {
-        return Flux.fromStream(limit.applyOnStream(messageIds.stream().distinct())
-            .collect(JamesCollectors.chunker(configuration.getMessageReadChunkSize())))
-            .flatMap(ids -> rowToMessages(fetchType, ids));
-    }
-
-    private Flux<MessageResult> rowToMessages(FetchType fetchType, Collection<ComposedMessageIdWithMetaData> ids) {
-        return Flux.fromIterable(ids)
+        return Flux.fromStream(limit.applyOnStream(messageIds.stream().distinct()))
```

Comment:
You probably won't have the concurrency you want because I don't provide a Scheduler (and by default, it runs in the calling thread with a concurrency of 1)

Reply:
I don't really get what you mean: `Flux.fromIterable(...)` -> operations will be sequential. Is that it?

I saw the exact same code snipset in `CassandraMessageIdMapper::delete`. What's different here?

Reply:
`fromIterable` doesn't give you any guarantee about ordering at all and flatMap definitely screws the ordering (there's a flatMapSequential to keep ordering).
What I'm saying is that limitRate controls the "pipelining depth" (how many concurrent items are traversing the pipeline at the same time). But if you have a one-thread scheduler, you will have only a single item whatever the rate you ask for.

Reply:
1. We don't care about ordering because a final UID ordering step is performed here

2. Do you want me to run a quick gatling bench to sort the 'perf' issue out?